### PR TITLE
[Silabs]- Adds fix for failed last connect error value

### PR DESCRIPTION
**Pull request by LeelakumarPS**
* Adds fix for failed last connect error value
* Tested manually

### CODE REVIEW

**Review by rosahay-silabs — APPROVED** [review]

File: src/platform/silabs/NetworkCommissioningWiFiDriver.cpp [hunk 1]
```diff
@@ -179,7 +179,8 @@ void SlWiFiDriver::UpdateNetworkingStatus()
     ByteSpan networkId = ByteSpan((const unsigned char *) mStagingNetwork.ssid, mStagingNetwork.ssidLen);
     if (!wfx_is_sta_connected())
     {
-        mpStatusChangeCallback->OnNetworkingStatusChange(Status::kUnknownError, MakeOptional(networkId), NullOptional);
+        mpStatusChangeCallback->OnNetworkingStatusChange(Status::kUnknownError, MakeOptional(networkId),
+                                                         MakeOptional((int32_t) SL_STATUS_FAIL));
```
Comment (bzbarsky-apple) [inline src/platform/silabs/NetworkCommissioningWiFiDriver.cpp:183]:
```suggestion
                                                         MakeOptional(static_cast<int32_t>(SL_STATUS_FAIL)));
```

**Review by bzbarsky-apple — APPROVED** [review]

File: src/platform/silabs/NetworkCommissioningWiFiDriver.cpp (hunk 1 above)
Comment (bzbarsky-apple) [reply src/platform/silabs/NetworkCommissioningWiFiDriver.cpp:183]: Actually, wait.  SL_STATUS_FAIL is 0x0001, right?

The value that's supposed to go here is:

> * On any association failure during enabling of a network, the ErrorValue field SHALL be set to the Status Code value that was present in the last frame related to association where Status Code was not equal to zero and which caused the failure of a final retry attempt, if this final failure was due to one of the following Management frames:
>
>   * Association Response (Type 0, Subtype 1)
>
>   * Reassociation Response (Type 0, Subtype 3)
>
>   * Authentication (Type 0, Subtype 11)
>
>    Table 9-50 "Status Codes" in IEEE 802.11-2020 contains a description of all values possible, which can unambiguously be used to determine the cause, such as an invalid security type, unsupported rate, etc.

This does not seem to be implementing that.

**Review by jepenven-silabs — CHANGES_REQUESTED** [review]
@bzbarsky-apple comment should be addressed.

File: src/platform/silabs/NetworkCommissioningWiFiDriver.cpp (hunk 1 above)
Comment (LeelakumarPS) [reply src/platform/silabs/NetworkCommissioningWiFiDriver.cpp:183]: @bzbarsky-apple 
Yes , SL_STATUS_FAIL is 0x0001
all the places it is returning a generic failure state

Comment (bzbarsky-apple) [reply src/platform/silabs/NetworkCommissioningWiFiDriver.cpp]: >  all the places it is returning a generic failure state

As in, all our implementations of this are buggy?

Comment (rosahay-silabs) [reply src/platform/silabs/NetworkCommissioningWiFiDriver.cpp]: @bzbarsky-apple as discussed, I opened up #26861 lets move forward with generic error approach till we figure out how to get the data from our driver.

**Review by bzbarsky-apple — APPROVED** [review]
Please add a comment pointing to the issue.

**Comment by LeelakumarPS** [comment]
align WiFi network join failure "Status Codes" in IEEE 802.11-2020 #26861 [https://github.com/project-chip/connectedhomeip/issues/26861](url)

**Review by jmeg-sfy — APPROVED** [review]

File: src/platform/silabs/NetworkCommissioningWiFiDriver.cpp [hunk 2]
```diff
@@ -179,7 +179,8 @@ void SlWiFiDriver::UpdateNetworkingStatus()
     ByteSpan networkId = ByteSpan((const unsigned char *) mStagingNetwork.ssid, mStagingNetwork.ssidLen);
     if (!wfx_is_sta_connected())
     {
-        mpStatusChangeCallback->OnNetworkingStatusChange(Status::kUnknownError, MakeOptional(networkId), NullOptional);
+        mpStatusChangeCallback->OnNetworkingStatusChange(Status::kUnknownError, MakeOptional(networkId),
```
Comment (jepenven-silabs) [inline src/platform/silabs/NetworkCommissioningWiFiDriver.cpp:182]: @LeelakumarPS The comment for the github issue should be place in the code.

Comment (LeelakumarPS) [reply src/platform/silabs/NetworkCommissioningWiFiDriver.cpp:183]: Added a comment  mentioning github issue

**Review by jmeg-sfy — APPROVED** [review]

**Review by jepenven-silabs — APPROVED** [review]

**Pull request merged**